tests/video_pipeline_test.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+import multiprocessing as mp
+from modules.targetAcquisition.targetAcquisitionWorker import targetAcquisitionWorker
+from modules.decklinksrc.decklinkSrcWorker import decklinkSrcWorker
+
+
+# Integration test to check mediator behaviour
+def testVideoPipeline():
+    videoPipeline = mp.Queue()
+    coordinatePipeline = mp.Queue()
+    pause = mp.Lock()
+    exitSignal = mp.Queue()
+
+    processes = [
+        mp.Process(target=decklinkSrcWorker, args=(pause, exitSignal, videoPipeline)),
+        mp.Process(target=targetAcquisitionWorker, args=(pause, exitSignal, videoPipeline, coordinatePipeline))
+    ]
+
+    for p in processes:
+        p.start()
+
+    assert coordinatePipeline.get() is None
+
+    exitSignal.put(True)
+
+
+def testPipelinePause():
```

Comment:
This test is kind of iffy tbh, it assumes that the qsize and pause happen almost instantaneously, which seems to be the case most of the time, maybe we can just get rid of it, or look at a way of checking when a process is blocked. Not too too high priority.

Reply:
Having race conditions is bad. We can reduce the chance though.

Consider this approach: pause, wait some time, get the size, wait some time, get the size and assert.

Unfortunately there aren't really any good ways of testing multiprocesses systems. In C++, Helgrind is one such system, but I don't know any for Python.

Reply:
It may not be worth testing multiprocess specifically because of non-determinism.

Reply:
Yeah, fuck it lol